### ISSUE TITLE
Can not set keybinds to non-letter keys

### ISSUE THREAD

**Issue by schockocraft** [widelands/widelands#5915]
**Describe the bug**
I can not set keybinds to any non-letter keys. 

**To reproduce**
1. Go to 'Options' > 'Edit keyboard and mouse actions...'
2. Click on the button of the keybind to change (in my case for 'Pause')
3. With the "Press the new shortcut (...)" window open, press space.
4. Nothing happens, the game still waits for me to press a key.

**Expected behavior**
The Game registers the key press and changes the keybind accordingly.

**Version:**
 - OS: Windows 10 Pro 21H2, Build 19044.2364
 - Widelands Version: Release v1.1

**Additional context**
I want to have pause on space bar, as i'm used to that from other games like cities skylines and banished, but it doesn't let me.

I also tried Shift, Ctrl, Alt and Enter keys, none of which are assignable to keybinds.
(for ctrl it makes sense tho as other keybinds use it as a modifier key)

**Comment by tothxa** [comment]
I've came across this too while testing a recent PR, and found that it's actually deliberate. I believe space and enter are excluded because they are used to trigger UI elements like buttons.

**Comment by Noordfrees** [comment]
Certain system keys are reserved and cannot be set as shortcuts:
https://github.com/widelands/widelands/blob/e51951d09fee70460f3b0d23fd5896a3cd01f5d6/src/ui_fsmenu/keyboard_options.cc#L166-L185

But instead of silently ignoring the keypress we should inform the user that these keys may not be used

**Comment by tothxa** [comment]
> But instead of silently ignoring the keypress we should inform the user that these keys may not be used

That, plus I think we could allow them together with modifiers. Except maybe ctrl alone, because we use that in an unconfigurable way for too many things that may affect at least arrows, tab and space IIRC.

**Comment by schockocraft** [comment]
> Certain system keys are reserved and cannot be set as shortcuts:
> 
> https://github.com/widelands/widelands/blob/e51951d09fee70460f3b0d23fd5896a3cd01f5d6/src/ui_fsmenu/keyboard_options.cc#L166-L185
> 
> But instead of silently ignoring the keypress we should inform the user that these keys may not be used

That is a major bummer, especially as enter and space are already bound in the default keybinds, so it doesn't even make sense.

I really don't think there's any good reason not to allow you to rebind space/enter, and also some people might prefer to move the view with wasd instead of arrow keys. I hoped to set tab to the "cycle to notification POIs" functionality, and alt to open the build menu (if you can open it before choosing a building location, i didn't get that far yet because i got annoyed about the keybinds xD).

I can understand the modifier keys being excluded as that would require additional logic though.

It would be cool if theres an override setting for that kinda rules, but if not then i guess i'll finally make myself learn Autohotkey.

At least it should be mentioned in the keybind settings.

Also i think there definitely should be a way to unset certain keybinds without adding another key to the function, tho that might belong into a separate issue?

**Comment by schockocraft** [comment]
Actually what about these keys is "reserved"? Many games and Programs i use bind these...

**Comment by tothxa** [comment]
> Actually what about these keys is "reserved"? Many games and Programs i use bind these...

Mostly that there are still some places, where the keys are hardcoded. These are usually basic interaction with UI elements.

**Comment by schockocraft** [comment]
> > Actually what about these keys is "reserved"? Many games and Programs i use bind these...
> 
> Mostly that there are still some places, where the keys are hardcoded. These are usually basic interaction with UI elements.

Oh, so it's just a temporary workaround until that's "fixed"?

**Comment by tothxa** [comment]
> Oh, so it's just a temporary workaround until that's "fixed"?

Well, it wasn't meant as temporary, otherwise there wouldn't be extra code to prevent assigning them. I added the "under discussion" tag because your request is an argument for overriding that decision. OTOH Widelands uses a windows/menus/buttons user interface, so it tries to follow common key conventions with other similar programs, not with fast paced action games, so assigning pure modifiers will very likely never be possible. But I agree with you that we shouldn't actively prevent users from setting keys to their liking, especially not those that we already assign by default to in-game actions (like space and enter).
